core/resources.rs

```diff
@@ -42,11 +43,14 @@ impl ResourceTable {
     None
   }
 
-  // TODO: resource id allocation should probably be randomized for security.
+  // resource id allocation are randomized for security.
   fn next_rid(&mut self) -> ResourceId {
-    let next_rid = self.next_id;
-    self.next_id += 1;
-    next_rid as ResourceId
+    let mut rng = rand::thread_rng();
+    let mut next_rid = rng.gen::<u32>() & 0x7FFF_FFFF;
```

Comment:
I guess ` 0x7FFF_FFFF` is the largest integer allowed in JS? I think this deserves a comment.

Reply:
In fact, no. But rid's are cast into i32 then interpreted as unsigned number on typescript side.
Then when sending rid's back to Rust the cast back to i32 fails as it overflows.
There are two options :
- the one i took and make the generated rid a valid positive i32
- make rid u32 on Ops api side. But there might be a good reason someone made them i32, and i did not have the Time to investigate why

Reply:
i actually need feedback on this, so i'll not resolve